GitCommands/CommitDataManager.cs

```diff
@@ -204,7 +204,7 @@ public void UpdateBodyInCommitData(CommitData commitData, string data)
             string message = ProccessDiffNotes(startIndex, lines);
 
             // commit message is not reencoded by git when format is given
-            Debug.Assert(commitData.Guid == guid);
+            Debug.Assert(commitData.Guid == guid, "commitData.Guid == guid");
```

Comment:
in this case, is there a value in this rule at all?

Reply:
I've found it can be helpful to see the expression in the assertion window, though you can often reverse engineer the assertion's condition from the stack trace. If a user ran a debug build without PDB files, then it might disambiguate between multiple assertions in the same method. But that's an unlikely situation. Totally happy to remove this rule's commit. Just let me know.

Reply:
Thanks, makes sense, let's keep it

Reply:
My preference is to put the actual values into the message:
`Debug.Assert(commitData.Guid == guid, "${commitData.Guid} != ${guid}");`
I would also opt for making assertions to fail in the production code as well.

Reply:
@jbialobr just seen this. Good idea, re: inserting values. To have assertions run in production code they'd have to be changed to [`Trace.Assert`](https://msdn.microsoft.com/en-us/library/tk4kachs(v=vs.110).aspx) calls, as `Debug.Assert` is `[Conditional("DEBUG")]`.